inception-search-core/src/main/java/de/tudarmstadt/ukp/inception/search/scheduling/TaskConsumer.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2017
+ * Ubiquitous Knowledge Processing (UKP) Lab
+ * Technische Universität Darmstadt
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.tudarmstadt.ukp.inception.search.scheduling;
+
+import static org.apache.commons.lang3.Validate.notNull;
+
+import java.util.concurrent.BlockingQueue;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.config.AutowireCapableBeanFactory;
+import org.springframework.context.ApplicationContext;
+
+import de.tudarmstadt.ukp.inception.search.scheduling.tasks.Task;
+
+public class TaskConsumer
+    implements Runnable
+{
+    private Logger log = LoggerFactory.getLogger(getClass());
```

Comment:
I would make logs static if there is no good reason not to.

Reply:
OK. But it should be seen in other points of INCEpTION code, for most of them declare the log variable as non-static.

Reply:
Currently, log variables are only static when required in static methods or when used in Wicket components. Otherwise we keep them non-static. This SO post includes a few sensible answers explaining why one might always want to use static (https://stackoverflow.com/questions/6653520/why-do-we-declare-loggers-static-final). I usually prefer using non-static to avoid copy-paste problems. `log = LogFactory.getLog(getClass())` works when copy/pasted, but `log = LogFactory.getLog(MyClass.class)` is usually wrong when copy/pasted.